nix/checks.nix

```diff
@@ -13,12 +13,30 @@
         # todo this could be moved into a shared flake
         reuse =
           pkgs.runCommandLocal "reuse-lint" {
-            buildInputs = [pkgs.reuse];
+            nativeBuildInputs = [pkgs.reuse];
           } ''
             cd ${../.}
             reuse lint
             touch $out
           '';
+        # pycodestyle
+        pycodestyle =
+          pkgs.runCommandLocal "pycodestyle" {
+            nativeBuildInputs = [pkgs.python3.pkgs.pycodestyle];
+          } ''
+            cd ${../.}
+            pycodestyle --max-line-length 90 $(find . -name "*.py" ! -path "*venv*" ! -path "*eggs*")
+            touch $out
+          '';
+        # pylint
+        pylint =
+          pkgs.runCommandLocal "pylint" {
+            nativeBuildInputs = with pkgs.python3Packages; [colorlog gitpython pandas pylint pytest tabulate];
+          } ''
+            cd ${../.}
+            pylint --disable duplicate-code -rn $(find . -name "*.py" ! -path "*venv*" ! -path "*eggs*")
+            touch $out
+          '';
```

Comment:
We might be able to run these with treefmt. I think it can be configured to run arbitrary commands. Failing that, I might have a look at adding support directly for `pycodestyle` and `pylint`.

Reply:
@brianmcgee : thanks for you comment. The changes in this PR are based on Florian's changes in: https://github.com/tiiuae/sbomnix/pull/94/files#diff-49f7a3ba809188883d2a60aef5bcd93529587f91ebb40004b7fc3f412feae79cR22-R38.

I propose we would merge this now, and then do another round of changes (to other repositories too) to move them to `treefmt`, if that's possilble.